src/user/dto/job-creation.dto.ts

```diff
@@ -1,30 +1,52 @@
-import { ApiProperty } from '@nestjs/swagger';
-import { IsNotEmpty, IsOptional, IsUrl } from 'class-validator';
+import { IsNotEmpty, IsOptional, IsDefined } from "class-validator";
+import { Company, CareerProfile, UserProfile } from "@prisma/client";
 
 /**
  * Creates a new Job
  */
 export class JobCreationDto {
-  @IsNotEmpty()
-  jobtitle: string;
-  @IsNotEmpty()
-  starttime: Date;
-  @IsNotEmpty()
-  endtime: Date;
-  @IsNotEmpty()
-  companyId:string;
-  @IsOptional()
-  userId:string;
-
-  @IsOptional()
-  jobIdAtBerufeNet? : string;
-
-  constructor(jobtitle: string, starttime: Date, endtime: Date, companyId: string, userId: string, jobIdAtBerufeNet? :string) {
-    this.jobtitle = jobtitle;   
-    this.starttime = starttime;
-    this.endtime = endtime;
-    this.companyId = companyId;
-    this.userId = userId;
-    this.jobIdAtBerufeNet = jobIdAtBerufeNet ?? undefined;
-  }
-}
\ No newline at end of file
+    @IsDefined()
+    id: string;
+
+    @IsNotEmpty()
+    jobtitle: string;
+
+    @IsNotEmpty()
+    starttime: Date;
+
+    @IsNotEmpty()
+    endtime: Date;
+
+    @IsOptional()
+    company?: Company;
+
+    @IsNotEmpty()
+    companyId: string;
+
+    @IsOptional()
+    userJobs?: CareerProfile[];
+
+    user: UserProfile;
```

Comment:
`JobCreationDto` is a request object, i.e., sent from outside and expects a `user` (internal `Prisma` object), which cannot send from the outside. Please resolve.